dask/array/routines.py

```diff
@@ -836,20 +835,26 @@ def ravel(array):
 
 @wraps(np.squeeze)
 def squeeze(a, axis=None):
-    if 1 not in a.shape:
-        return a
     if axis is None:
         axis = tuple(i for i, d in enumerate(a.shape) if d == 1)
-    b = a.map_blocks(partial(np.squeeze, axis=axis), dtype=a.dtype)
-    chunks = tuple(bd for bd in b.chunks if bd != (1,))
+    else:
```

Comment:
Do we need this `else` here?

Reply:
Yes, this handles both `int` and `tuple`s of `int`s. It coerces the former into the latter first (maybe part of the confusion). Then applies some checks and normalizes the input data. None of these checks or normalization are required for `None` as the data is already constructed to meet our expectations.

Reply:
I agree that the body of the else is useful.  I think we should the normalization code regardless, just to reduce the code nesting.

Reply:
Sure, fixed. :)